aliza/server.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+const http = require('http');
+
+const database = {};
+
+http.createServer((req, res) => {
+  if (req.url === '/time' && req.method === 'GET'){
+    let date = new Date();
+    let time = date.getHours() + ':' + date.getMinutes();
+    res.write(time + '\n');
+    return res.end();
+  }
+
+  let user = req.url.substring(7);
+  if (req.url === '/greet/' + user && req.method === 'GET'){
+    res.write('Hello ' + user + '\n');
+    return res.end();
+  }
+
+  //curl localhost:3000/greet -X POST -d '{"name":"aliza"}'
+
+  if (req.url === '/greet' && req.method === 'POST') {
+    var name = '';
+    req.on('data', (data) => {
+      name += data.toString();
+    });
+    req.on('end', () => {
+      var nameObj = JSON.parse(name);
+      res.write('Hello ' + nameObj.name + '\n');
+      return res.end();
+    })
```

Comment:
If you wanted to move to away from the if/then/else you would put a return after the last handler was registered (line 33).  The idea is the your route assigns handlers and then leaves the create server code.  Its done everything it needs to do by assigning the handlers.  The return within the asynch handler code isn't really necessary since there is no code to execute in after it.  POST routers rely on data that won't be available synchronously, so all they can do is register handler callbacks and then get out the of create server code.  They don't want to end the res stream.